lib/Models/upsertModelFromJson.ts

```diff
@@ -29,16 +29,12 @@ export default function upsertModelFromJson(
         });
       }
 
-      let id = (parentId || "") + "/" + localId;
-      let idIncrement = 1;
-      uniqueId = id;
-
-      while (terria.getModelById(BaseModel, uniqueId) !== undefined) {
-        uniqueId = id + "(" + idIncrement + ")";
-        idIncrement++;
-      }
+      uniqueId = (parentId || "") + "/" + localId;
     }
 
+    // To handle duplicate items, e.g. same item in different tabs
+    uniqueId = incrementIfAlreadyExists(uniqueId, terria);
+
```

Comment:
The reason this breaks the  [TerriaSpec](https://github.com/TerriaJS/terriajs/blob/next/test/Models/TerriaSpec.ts#L248) is that terria automatically creates a group with id [USER_ADDED_CATEGORY_ID.](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/CatalogNew.ts#L38). When [updateApplicationUrl](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/Terria.ts#L591) is called with a share link containing `USER_ADDED_CATEGORY_ID` group, the above change will auto increment and create `USER_ADDED_CATEGORY_ID (1)` etc.